runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1404,3 +1404,44 @@ TR_JITServerRelocationRuntime::copyDataToCodeCache(const void *startAddress, siz
    return coldCodeStart;
    }
 #endif /* defined(J9VM_OPT_JITSERVER) */
+
+void
+printAOTHeader(const void* aotHeaderAddress, char * buff, size_t buffSize)
+   {
+   if (!aotHeaderAddress)
+      {
+      strncat(buff, "null", buffSize);
+      return;
+      }
+
+   PORT_ACCESS_FROM_PORT(TR::Compiler->portLib);
+   TR_AOTHeader * hdrInCache = (TR_AOTHeader *)aotHeaderAddress;
+
+   OMRProcessorDesc processorDescription = hdrInCache->processorDescription;
+   OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
+   int length = 0;
+   for (size_t i = 0; i < OMRPORT_SYSINFO_FEATURES_SIZE; i++)
+      {
+      for (int j = 0; j < 32; j++) 
+         {
+         if (processorDescription.features[i] & (1<<j))
+            {
+            if (length >= 20)
+               {
+               strncat(buff, "\n\t                                       ", buffSize);
```

Comment:
I think you should keep track of return value of `strncat` so that you don't overrun `buff`.

Reply:
Would also be better to use the size of the source string, rather than `buffSize` which is the size of the dest.

Reply:
I intended to use the third parameter to prevent overflow but I didn't do it right
Should be something like this:
The third parameter gives you the maximum number of chars allowed to be written to buff.
```cpp
strncat(buff, "string", BUFF_SIZE - strlen(buff) - 1);
```
I don't think the return value of `strncat` help here.